simulation/tests/test_costs.py

```diff
@@ -174,3 +175,63 @@ def test_air_source_heat_pumps_get_cheaper_across_2022(self):
                 assert quote == future_quote
 
             quote = future_quote
+
+    @pytest.mark.parametrize("boiler", set(BOILERS))
+    def test_boiler_upgrade_scheme_grant_is_always_zero_for_boilers(self, boiler):
+
+        start_datetime = datetime.datetime(2022, 1, 1, 0, 0)
+        end_datetime = datetime.datetime(2032, 1, 1, 0, 0)
+        random_n_days = random.randrange((end_datetime - start_datetime).days)
+        start_datetime = start_datetime + datetime.timedelta(days=random_n_days)
+
+        model = model_factory(
+            start_datetime=start_datetime,
+        )
+
+        assert estimate_boiler_upgrade_scheme_grant(boiler, model) == 0
+
+    @pytest.mark.parametrize("heat_pump", set(HEAT_PUMPS))
```

Comment:
I suggest this test tests that the scheme is zero outside the window for _all_ heating systems. What do you think?
```suggestion
    @pytest.mark.parametrize("heating_system", set(HeatingSystem))
```

Reply:
Good idea